Userland/Libraries/LibJS/Runtime/Completion.h

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright (c) 2021, Andreas Kling <kling@serenityos.org>
+ * Copyright (c) 2021, Linus Groh <linusg@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/FlyString.h>
+#include <AK/Optional.h>
+#include <AK/Try.h>
+#include <LibJS/Runtime/Value.h>
+
+namespace JS {
+
+// Temporary helper akin to TRY(), but returning a default-constructed type (e.g. empty JS::Value)
+// instead of the throw completion record. Use this as the bridge between functions that have
+// already been updated to use completions and functions that haven't.
+#define TRY_OR_DISCARD(expression)  \
+    ({                              \
+        auto result = (expression); \
+        if (result.is_error())      \
+            return {};              \
+        result.release_value();     \
+    })
+
+// 6.2.3 The Completion Record Specification Type, https://tc39.es/ecma262/#sec-completion-record-specification-type
+class Completion {
```

Comment:
Can we mark this entire class `[[nodiscard]]`? That way we don't need to say `[[nodiscard]]` when it's being used as a return type, it will always be implied.

Reply:
Agreed, done :+1: